client/foundries_sboms.go

```diff
@@ -39,18 +40,26 @@ func (a *Api) TargetSboms(factory string, targetName string) ([]Sbom, error) {
 	}
 	for idx := range sboms {
 		sboms[idx].api = a
+		sboms[idx].factory = factory
 	}
 	return sboms, nil
 }
 
-func (s Sbom) Download(dst string) error {
+func (s Sbom) Download(targetName, dst string, asCsv bool) error {
+	if asCsv {
+		path := fmt.Sprintf("%s/%s/%s", s.CiBuild, s.CiRun, s.Artifact)
+		buf, err := s.api.TargetSbomPackagesCsv(s.factory, targetName, path)
+		if err != nil {
+			return err
+		}
+		return os.WriteFile(dst, buf, 0o744)
+	}
```

Comment:
IMHO this thing becomes hard to understand.

For example, I don't understand why we download from `s.Uri` in one case, and craft a different Uri in another one; I had to look into the API implementation to understand that.

So, two comments here:

1. It would be nice to have a comment why we download from Uri in one case, and do Api call in another case.
2. I think a cleaner solution would be to use the `api.Get(sbom.URI), api.TargetSbomPackagesCsv(factory, targetName, sbom.ApiPath())` directly inside the `--download` option handler; that would keep complexity in just one place.